src/murmur/Messages.cpp

```diff
@@ -1391,30 +1396,44 @@ void Server::msgTextMessage(ServerUser *uSource, MumbleProto::TextMessage &msg)
 	}
 
 	msg.set_actor(uSource->uiSession);
+
+	// Send the message to all users that are in (= have joined) OR are
+	// "listening" to channels to which the message has been directed to
 	for (int i = 0; i < msg.channel_id_size(); ++i) {
 		unsigned int id = msg.channel_id(i);
 
 		Channel *c = qhChannels.value(id);
-		if (!c)
+		if (!c) {
 			return;
+		}
 
 		if (!ChanACL::hasPermission(uSource, c, ChanACL::TextMessage, &acCache)) {
 			PERM_DENIED(uSource, c, ChanACL::TextMessage);
 			return;
 		}
 
-		foreach (User *p, c->qlUsers)
+		// Users directly in that channel
+		foreach (User *p, c->qlUsers) {
 			users.insert(static_cast< ServerUser * >(p));
+		}
+
+		// Users only listening in that channel
+		foreach (unsigned int session, ChannelListener::getListenersForChannel(c)) {
+			users.insert(qhUsers.value(session));
```

Comment:
```suggestion
			ServerUser *currentUser = qhUsers.value(session);
			if (currentUser) {
				users.insert(qhUsers.value(session));
			}
```

This just makes sure that we are not accidentally producing a `nullptr` that gets fed into `users`

Reply:
Indeed, valid comment. I've fixed that :)